File: j-LIB-core/src/main/java/io/github/dot166/jlib/app/jConfigActivity.java
package io.github.dot166.jlib.app;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;

import androidx.annotation.Nullable;
import androidx.core.view.WindowCompat;
import androidx.preference.Preference;
import androidx.preference.PreferenceFragmentCompat;
import androidx.preference.PreferenceScreen;

import java.util.Objects;

import io.github.dot166.jlib.R;
import io.github.dot166.jlib.LIBAboutActivity;
import io.github.dot166.jlib.utils.VersionUtils;

/**
 * jLib Settings activity.
 */
public class jConfigActivity extends jActivity {

    public PreferenceFragmentCompat preferenceFragment() {
        return new PreferenceFragmentCompat() {
            @Override
            public void onCreatePreferences(@Nullable Bundle savedInstanceState, @Nullable String rootKey) {
                PreferenceScreen screen = getPreferenceScreen();
                Preference disclaimer = new Preference(requireContext());
                disclaimer.setTitle("This is an example preference screen, please override the preferenceFragment() function to use your own preference fragment");
                screen.addPreference(disclaimer);
                Preference libPref = new Preference(requireContext());
                libPref.setIcon(android.R.drawable.sym_def_app_icon);
                libPref.setTitle(R.string.jlib_version);
                libPref.setSummary(String.valueOf(VersionUtils.getLibVersion()));
                libPref.setOnPreferenceClickListener(preference -> {
                    startActivity(new Intent(preference.getContext(), LIBAboutActivity.class));
                    return true;
                });
                screen.addPreference(libPref);
            }
        };
    }

    @SuppressLint("PrivateResource")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.settings_activity);
        setSupportActionBar(findViewById(R.id.actionbar));
        Objects.requireNonNull(getSupportActionBar()).setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setHomeActionContentDescription(androidx.appcompat.R.string.abc_action_bar_up_description);

        WindowCompat.setDecorFitsSystemWindows(getWindow(), false);

        if (savedInstanceState == null) {
            // Display the fragment as the main content.
            getSupportFragmentManager().beginTransaction().replace(R.id.content_frame, preferenceFragment()).commit();
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            getOnBackPressedDispatcher().onBackPressed();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: j-LIB-core/src/main/java/io/github/dot166/jlib/app/jConfigActivity.kt
package io.github.dot166.jlib.app

import android.annotation.SuppressLint
import android.app.ActivityOptions
import android.content.Intent
import android.os.Bundle
import android.view.Display
import androidx.fragment.app.Fragment
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import com.android.settingslib.collapsingtoolbar.CollapsingToolbarBaseActivity
import com.android.settingslib.preference.PreferenceFragment

/**
 * jLib Settings activity.
 */
abstract class jConfigActivity : CollapsingToolbarBaseActivity(), PreferenceFragmentCompat.OnPreferenceStartFragmentCallback {
    abstract fun preferenceFragment(): PreferenceFragment

    @SuppressLint("PrivateResource")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        supportFragmentManager.beginTransaction()
            .replace(
                com.android.settingslib.collapsingtoolbar.R.id.content_frame,
                preferenceFragment()
            ).addToBackStack(null).commit()
    }

    override fun startActivity(intent: Intent) {
        val options = ActivityOptions.makeBasic()
        options.setLaunchDisplayId(Display.DEFAULT_DISPLAY) // Forces it to the main screen
        startActivity(intent, options.toBundle())
    }

    override fun startActivityForResult(intent: Intent, requestCode: Int) {
        val options = ActivityOptions.makeBasic()
        options.setLaunchDisplayId(Display.DEFAULT_DISPLAY) // Forces it to the main screen
        startActivityForResult(intent, requestCode, options.toBundle())
    }

    override fun onPreferenceStartFragment(
        caller: PreferenceFragmentCompat,
        pref: Preference
    ): Boolean {
        val args: Bundle = pref.getExtras()
        val fragment: Fragment = supportFragmentManager.getFragmentFactory().instantiate(
            classLoader, pref.fragment!!
        )
        fragment.setArguments(args)
        supportFragmentManager.beginTransaction().setCustomAnimations(
            android.R.anim.slide_in_left,
            android.R.anim.slide_out_right,
            android.R.anim.slide_in_left,
            android.R.anim.slide_out_right
        )
            .replace(com.android.settingslib.collapsingtoolbar.R.id.content_frame, fragment)
            .addToBackStack(null).commit()
        return true
    }
}
